fdmi/plugins/sched.c

```diff
@@ -422,79 +407,84 @@ static int fdmi_service_start(struct m0_client *m0c){
 		m0_reqh_service_init(sched_fdmi_service, reqh, NULL);
 		start_service = true;
 	}
 
-	if (start_service) {
-		rc = m0_reqh_service_start(sched_fdmi_service);
-		M0_POST(rc == 0);
-	}
+	if (start_service)
+        	rc = m0_reqh_service_start(sched_fdmi_service);
 	return M0_RC(rc);
 }
 
+static void fdmi_service_stop(struct m0_client *m0c)
+{
+	if (sched_fdmi_service != NULL) {
+		m0_reqh_service_stop(sched_fdmi_service);
+		sched_fdmi_service = NULL;
+	}
+}
+
 static int init_fdmi_plugin(struct sched_conf *conf)
 {
-    int rc;
+	int rc;
 
-    pdo = m0_fdmi_plugin_dock_api_get();
-    const struct m0_fdmi_filter_desc fd;
+	pdo = m0_fdmi_plugin_dock_api_get();
+	const struct m0_fdmi_filter_desc fd;
 
-    const static struct m0_fdmi_plugin_ops pcb = {
-        .po_fdmi_rec = classify_handle_fdmi_rec_not
-    };
+	const static struct m0_fdmi_plugin_ops pcb = {
+		.po_fdmi_rec = classify_handle_fdmi_rec_not
+	};
 
-    /* printf("Registering classify plugin."); */
-    rc = pdo->fpo_register_filter(&conf->fdmi_plugin_fid, &fd, &pcb);
-    printf("Plugin registration rc: %d\n", rc);
-    if (rc < 0)
-        goto end_fdmi_init;
+	rc = pdo->fpo_register_filter(&conf->fdmi_plugin_fid, &fd, &pcb);
+	fprintf(stderr, "Plugin registration failed: rc=%d\n", rc);
+	if (rc != 0)
+		return rc;
 
-    /* printf("Classify rc: %d", rc); */
-    pdo->fpo_enable_filters(true, &conf->fdmi_plugin_fid, 1);
- end_fdmi_init:
-    return rc;
+	pdo->fpo_enable_filters(true, &conf->fdmi_plugin_fid, 1);
+	return rc;
 }
-static void deinit_plugin(struct sched_conf *conf)
+
+static void fini_fdmi_plugin(struct sched_conf *conf)
 {
 	pdo->fpo_enable_filters(false, &conf->fdmi_plugin_fid, 1);
 	pdo->fpo_deregister_plugin(&conf->fdmi_plugin_fid, 1);
-
 }
 
-#include <unistd.h>
-#include <fcntl.h>
+#if 0
+static int write_plugin_endpoint(char *fdmi_plugin_ep) {
+	int fd;
+	int rc;
 
-#define FDMI_PLUGIN_EP "/etc/fdmi_plugin_ep"
+	fd = open(FDMI_PLUGIN_EP, O_CREAT | O_WRONLY, 0666);
+	if (fd < 0) {
+		fprintf(stderr, "Failed to open %s: error=%d\n", FDMI_PLUGIN_EP, errno);
+		return errno;
+	}
 
-static void write_plugin_endpoint(char *fdmi_plugin_ep) {
-    int fd;
-    int rc;
-    fd = open(FDMI_PLUGIN_EP, O_CREAT | O_WRONLY, 0666);
-    if (fd < 0) {
-        m0_console_printf("open failed in write_plugin_endpoint\n");
-        return;
-    }
-
-    rc = write(fd, fdmi_plugin_ep, strlen(fdmi_plugin_ep));
-    if (rc < 0) {
-        m0_console_printf("write failed in write_plugin_endpoint\n");
-        return;
-    }
-    m0_console_printf("%s written to %s\n", fdmi_plugin_ep, FDMI_PLUGIN_EP);
-    close(fd);
+	rc = write(fd, fdmi_plugin_ep, strlen(fdmi_plugin_ep));
+	if (rc != 0) {
+		fprintf(stderr, "Failed to write to %s\n", FDMI_PLUGIN_EP);
+		close(fd);
+		return rc;
+	}
+	fprintf(stderr, "%s written to %s\n", fdmi_plugin_ep, FDMI_PLUGIN_EP);
+	close(fd);
+	return rc;
 }
+#endif
 
 static int sched_init(struct sched_conf *conf)
```

Comment:
With above suggestion to change it to be generic fdmi app, this can change to something like `fdmiapp_init()` ?

Reply:
Additionally for all functions in the code, please add comments at the start of function on what it does.

Reply:
ok